### PR TITLE
PendingStateManager: Throw fatal error rather than calling closeFn

### PR DESCRIPTION
**Pull request by markfields**
## Description

The error will propagate up to DeltaQueue which will emit error and close container.
We shouldn't continue processing in these cases - better to bail immediately.  Same flow as asserts, btw.

Also, I'm adding a fairly comprehensive log to tell us as much info as possible about the pending/incoming message when their stringified contents doesn't match, since we have seen this a few times in production.

### CODE REVIEW

File: packages/runtime/container-runtime/src/pendingStateManager.ts [hunk 1]
```diff
@@ -399,17 +398,14 @@ export class PendingStateManager implements IDisposable {
...
-						{
-							expectedMessageType: JSON.parse(pendingMessage.content).type,
-						},
-					),
+				throw DataProcessingError.create(
+					"pending local message content mismatch",
+					"unexpectedAckReceived",
+					message,
+					{
+						expectedMessageType: JSON.parse(pendingMessage.content).type,
```
Comment (kian-thompson) [inline packages/runtime/container-runtime/src/pendingStateManager.ts:406]:
```suggestion
						messageType: message.type,
						expectedMessageType: JSON.parse(pendingMessage.content).type,
```
Would also be nice to have some extra telemetry

File: packages/runtime/container-runtime/src/pendingStateManager.ts [hunk 2]
```diff
@@ -72,7 +71,6 @@ export type PendingMessageResubmitData = Pick<
 export interface IRuntimeStateHandler {
 	connected(): boolean;
 	clientId(): string | undefined;
-	close(error?: ICriticalContainerError): void;
```
Comment (rajatch-ff) [inline packages/runtime/container-runtime/src/pendingStateManager.ts:75]: >close(error?: ICriticalContainerError): void; [](http://example.com/codeflow?start=1&length=45)

Just for my knowledge - how do we ensure that removing something from an exported interface, wouldn't break any consumers? Where all do we search for consumers?

**Review by rajatch5 — APPROVED** [review]
:shipit:

File: packages/runtime/container-runtime/src/pendingStateManager.ts (hunk 2 above)
Comment (markfields) [reply packages/runtime/container-runtime/src/pendingStateManager.ts:75]: As long as there is no tag or only `@internal` tag, then the repo building is all the confirmation you need.

Comment (markfields) [reply packages/runtime/container-runtime/src/pendingStateManager.ts:75]: We are removing the type validation checks for things tagged `@internal` this sprint I believe.

**Review by kian-thompson — APPROVED** [review]

**Pull request merged**